### PR TITLE
Rails 6 support and fixes

### LINKED ISSUE #133: Rails 6 autoload issue
Upgrade to Rails 6.0

```
DEPRECATION WARNING: Initialization autoloaded the constants RecurringSelectHelper, and RecurringSelectMiddleware.

Being able to do this is deprecated. Autoloading during initialization is going
to be an error condition in future versions of Rails.

Reloading does not reboot the application, and therefore code executed during
initialization does not run again. So, if you reload RecurringSelectHelper, for example,
the expected changes won't be reflected in that stale Class object.

These autoloaded constants have been unloaded.

Please, check the "Autoloading and Reloading Constants" guide for solutions.
 (called from <top (required)> at /home/tim/rails/krtschmr/config/environment.rb:5)

```

### PR DESCRIPTION
**Pull request by uanka**
#133 

\- fix errors through the project
\- update specs
\- add support for rails 6 - move helper and middleware to lib/ as mentioned in rails/issues/336363
\- update Travis CI yml
\- update dummy app - remove redundant files and add post route to display params instead of routes error screen
\- make Gemfile clean (includes only gemspec)

### CODE REVIEW

**Comment by devanmoylan** [comment]
Thanks for doing this! Any update on when it might go in?

**Comment by jessevanherk** [comment]
code looks good, thanks for putting this up!

Due to a recent dependabot change there was a gemfile lock conflict - merging this to a temp branch where I can regen the lockfile before merging to master.

File: Gemfile.lock [hunk 1]
```diff
@@ -49,7 +49,7 @@ GEM
     execjs (2.7.0)
     hike (1.2.3)
     i18n (0.7.0)
-    ice_cube (0.16.2)
+    ice_cube (0.16.3)
```
Comment (jessevanherk) [inline Gemfile.lock:52]: this is the line that created the merge conflict

File: app/assets/javascripts/jquery-mobile-rs.js.coffee [hunk 2]
```diff
@@ -1,5 +1,5 @@
-//= require recurring_select
-//= require_self
+#= require recurring_select
+#= require_self
```
Comment (jessevanherk) [inline app/assets/javascripts/jquery-mobile-rs.js.coffee:2]: this was missed in #134 , nice catch!

File: app/assets/javascripts/recurring_select.js.coffee [hunk 3]
```diff
@@ -1,5 +1,5 @@
-//= require recurring_select_dialog
-//= require_self
+#= require recurring_select_dialog
+#= require_self
```
Comment (jessevanherk) [inline app/assets/javascripts/recurring_select.js.coffee:2]: this will close off #134

**Pull request merged**